### PR TITLE
Updates to Makefile and make.inc examples

### PR DESCRIPTION
**Pull request by abdelfattah83**
- For Makefile, Ada architecture (sm_89) is added, and sm_90a is added for Hopper. 
- Several make.inc examples are no longer relevant, like those using ACML, ATLAS, or those targeting macOS. Those are now removed. 
- Also replace icc/icpc/ifort with icx/icpx/ifx

### CODE REVIEW

**Comment by abdelfattah83** [comment]
@G-Ragghianti, I had to make minor changes in the workflows (make.inc file names with 'icc' now changed to 'icx'). Please double check.

**Comment by G-Ragghianti** [comment]
I had implemented a workaround for the change in compiler names by modifying the makefile includes.  This is done in setup.sh.  With your change, we should probably remove those lines.

**Comment by G-Ragghianti** [comment]
Looks like 'ifort' needs to be changed to 'ifx'.

**Comment by abdelfattah83** [comment]
@G-Ragghianti, two checks failed due to `icx`, `icpx`, and `ifx` being undefined. Can you please check?

File: .github/workflows/setup.sh [hunk 1]
```diff
@@ -18,14 +18,6 @@ else
    spack load intel-oneapi-mkl
 fi
 
-if [ "$COMPILER" = "intel" ]; then
-   spack load intel-oneapi-compilers
```
Comment (G-Ragghianti) [inline .github/workflows/setup.sh:22]: Still need to load the intel compilers

Comment (abdelfattah83) [reply .github/workflows/setup.sh]: Thanks, hopefully all checks should now pass

**Comment by G-Ragghianti** [comment]
OK looks good now.

**Pull request merged**